src/test/java/eu/europeana/metis/sandbox/service/dataset/DatasetServiceImplTest.java

```diff
@@ -55,7 +55,7 @@ void createDataset_expectSuccess() {
     when(datasetRepository.save(any(DatasetEntity.class))).thenReturn(datasetEntity);
     when(generatorService.generate("1", "name", Country.AUSTRIA, Language.BE, records))
         .thenReturn(dataset);
-    var result = service.createDataset("name", Country.AUSTRIA, Language.BE, records);
+    var result = service.createDataset("name", Country.AUSTRIA, Language.BE, records, null);
```

Comment:
I think those method calls should be without the null parameter at the end.
Also we probably want to test success and failure of the new method.

Reply:
I have removed the `null` parameters.
I have also added 2 new unit tests for the new `createDataset` method. The existing failure cases I believe already cover the new method.